library/std/src/sys/uefi/args.rs

```diff
@@ -0,0 +1,181 @@
+//! Args related functionality for UEFI. Takes a lot of inspiration of Windows args
+
+use super::common;
+use crate::ffi::OsString;
+use crate::fmt;
+use crate::num::NonZeroU16;
+use crate::os::uefi::ffi::OsStringExt;
+use crate::sys_common::ucs2::Ucs2Units;
+use crate::vec;
+use core::iter;
+use r_efi::efi::protocols::loaded_image;
+
+pub struct Args {
+    parsed_args_list: vec::IntoIter<OsString>,
+}
+
+// Get the Supplied arguments for loaded image.
+// Uses EFI_LOADED_IMAGE_PROTOCOL
+pub fn args() -> Args {
+    match common::get_current_handle_protocol::<loaded_image::Protocol>(loaded_image::PROTOCOL_GUID)
+    {
+        Some(x) => {
+            let lp_cmd_line = unsafe { (*x.as_ptr()).load_options as *const u16 };
+            let parsed_args_list =
+                parse_lp_cmd_line(unsafe { Ucs2Units::new(lp_cmd_line) }, || OsString::new());
+
+            Args { parsed_args_list: parsed_args_list.into_iter() }
+        }
+        None => Args { parsed_args_list: Vec::new().into_iter() },
+    }
+}
+
+/// Implements the Windows command-line argument parsing algorithm. Since UEFI is so similar, this
+/// can be used pretty much as is in UEFI
+///
+/// Microsoft's documentation for the Windows CLI argument format can be found at
+/// <https://docs.microsoft.com/en-us/cpp/cpp/main-function-command-line-args?view=msvc-160#parsing-c-command-line-arguments>
+///
+/// A more in-depth explanation is here:
+/// <https://daviddeley.com/autohotkey/parameters/parameters.htm#WIN>
+///
+/// This function was tested for equivalence to the C/C++ parsing rules using an
+/// extensive test suite available at
+/// <https://github.com/ChrisDenton/winarg/tree/std>.
+#[inline]
+fn parse_lp_cmd_line<'a, F: Fn() -> OsString>(
```

Comment:
This is the same as the function in windows/sys/args.rs right? Rather than duplicating it here, I think we should instead move that the function into `sys_common`, behind a cfg for windows/uefi.